Modules/_datetimemodule.c

```diff
@@ -7344,9 +7344,13 @@ init_static_types(PyInterpreterState *interp, int reloading)
 
     /* Bases classes must be initialized before subclasses,
      * so capi_types must have the types in the appropriate order. */
+    static PyMutex mutex = {0};
     for (size_t i = 0; i < Py_ARRAY_LENGTH(capi_types); i++) {
         PyTypeObject *type = capi_types[i];
-        if (_PyStaticType_InitForExtension(interp, type) < 0) {
+        PyMutex_Lock(&mutex);
+        int ret = _PyStaticType_InitForExtension(interp, type);
+        PyMutex_Unlock(&mutex);
```

Comment:
A lock needs to be applied to whole the loop or function rather than to a type to avoid a crash here (for me), because all [managed static extension types](https://github.com/python/cpython/blob/db47f4d844acf2b6e52e44f7f3d5f7566b1e402c/Include/internal/pycore_interp_structs.h#L532-L578
) are currently supposed to be indexed by using the same interpreter:

https://github.com/python/cpython/blob/db47f4d844acf2b6e52e44f7f3d5f7566b1e402c/Objects/typeobject.c#L322-L329

Reply:
> A lock needs to be applied to whole the loop or function

Hmm, still crashes with many workers...

Reply:
I agree, will update the code. You mean the crash mentioned in the original issue or something else? I found that with current change, the crash mentioned in the original issue (#136421) disappeared, but 136423 still exists.